internal/controller/clusters/cluster_controller.go

```diff
@@ -363,20 +352,21 @@ func (r *ClusterReconciler) SyncCleanup(ctx context.Context, logger logr.Logger,
 	}
 
 	if resp.StatusCode != http.StatusOK && obj.Code != "NotFound" {
```

Comment:
I feel like not returning here may not be ideal. If the user installed the vCluster through the extension UI, then this would cause it to leave an app that we created.